internal/tasks/subscriptiontask.go

```diff
@@ -236,6 +236,19 @@ func (sm *SubscriptionTask) handleNodeBindingUpdates(ctx context.Context, newBin
 						"subscription", newBinding.SubscriptionID,
 						"error", err.Error())
 				}
+
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					case restart := <-handler.GetRestartChannel():
+						if restart {
+							handler.Restart(ctx)
```

Comment:
Why send a false and true and check here? Send a signal on restartChan only if a restart is required.

Reply:
if a push stream was running but due to node binding change, we stop it, then we don't need to restart it. 
In that case, only push stream context is cancelled, and not the subscription task's.
To differentiate b/w these, had to send false on restart Chan to exit the start go routine